src/abe_policy/access_policy.rs

```diff
@@ -360,13 +360,15 @@ impl AccessPolicy {
                     .get(&attr.axis)
                     .ok_or_else(|| Error::AxisNotFound(attr.axis.to_string()))?;
                 let mut res = vec![vec![attr.clone()]];
-                if axis_parameters.is_hierarchical && follow_hierarchical_axes {
-                    // add attribute values for all attributes below the given one
-                    for name in &axis_parameters.attribute_names {
-                        if *name == attr.name {
-                            break;
+                if follow_hierarchical_axes {
+                    if let Some(order) = axis_parameters.order.as_deref() {
+                        // add attribute values for all attributes below the given one
+                        for name in order {
+                            if *name == attr.name {
+                                break;
+                            }
```

Comment:
You could make this a one-liner:
```suggestion
                        for name in order.iter().take_while(|&name| name != &attr.name) {
```